pallets/nfts/src/tests.rs

```diff
@@ -1903,6 +2157,140 @@ fn cancel_approval_works() {
 	});
 }
 
+#[test]
+fn cancel_collection_approval_works() {
+	new_test_ext().execute_with(|| {
+		let collection_id = 0;
+		let item_id = 42;
+		let collection_owner = account(1);
+		let item_owner = account(2);
+		let delegate = account(3);
+
+		Balances::make_free_balance_be(&item_owner, 100);
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			collection_owner.clone(),
+			default_collection_config()
+		));
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(collection_owner),
+			collection_id,
+			item_id,
+			item_owner.clone(),
+			default_item_config()
+		));
+
+		assert_ok!(Nfts::approve_collection_transfer(
+			RuntimeOrigin::signed(item_owner.clone()),
+			collection_id,
+			delegate.clone(),
+			None
+		));
+		assert_eq!(Balances::reserved_balance(&item_owner), 1);
+
+		// Cancel an unapproved delegate.
+		assert_noop!(
+			Nfts::cancel_collection_approval(
+				RuntimeOrigin::signed(item_owner.clone()),
+				1,
+				delegate.clone()
+			),
+			Error::<Test>::Unapproved
+		);
+
+		// Successfully cancel a collection approval.
+		assert_ok!(Nfts::cancel_collection_approval(
+			RuntimeOrigin::signed(item_owner.clone()),
+			collection_id,
+			delegate.clone()
+		));
+		assert_eq!(Balances::reserved_balance(&item_owner), 0);
+		assert!(events().contains(&Event::<Test>::ApprovalCancelled {
+			collection: collection_id,
+			item: None,
+			owner: item_owner.clone(),
+			delegate: delegate.clone()
+		}));
+		assert_eq!(
+			CollectionApprovals::<Test>::get((collection_id, item_owner, delegate.clone())),
+			None
+		);
+
+		assert_noop!(
+			Nfts::transfer(RuntimeOrigin::signed(delegate), collection_id, item_id, account(4)),
+			Error::<Test>::NoPermission
+		);
+	});
+}
+
+#[test]
+fn force_cancel_collection_approvals_work() {
```

Comment:
This code can combine with the `cancel_collection_approval_works` test to reduce code duplication by having a for loop over the `[(Runtime::signed(origin), None), (Runtime::root(), Some(owner))]` but it need additional helper method. I'm not sure if it is better as keeping them in a separate test case keep it simple, but increase maintenance effort.